extensions/replication/tasks/MultipleBackendTask.js

```diff
@@ -375,6 +376,9 @@ class MultipleBackendTask extends ReplicateObject {
         while (contentLen / rangeSize > 10000) {
             rangeSize *= 2;
         }
+        if (isGCP && (contentLen / rangeSize > MPU_GCP_MAX_PARTS)) {
+            rangeSize = this._getGCPRangeSize(contentLen);
```

Comment:
Sorry to insist, but wouldn't it be clearer if `_getGCPRangeSize` was computing the GCP range as a whole instead of being valid only for ranges > 512MB? Which brings to calling the generic `_getRangeSize` from inside, leaving the GCP logic to the GCP specialized function.

btw, "no" is a valid answer, just my personal though ;)

Reply:
Helpers can be reworked if it helps :)

Here's the outline of how I think the code would gain in clarity:

```js

function _getGenericRangeSize(contentLen) {
    // nothing related to GCP here
    return someSize;
}

function _getGCPRangeSize(contentLen) {
    range = _getGenericRangeSize(contentLen);
    if (/* range too big for GCP */) {
        // put here the GCP logic for ranges too big for GCP with standard computation
    } else {
        return range;
    }
}

if (isGCP) {
    _getGCPRangeSize(contentLen);
} else {
    _getGenericRangeSize(contentLen);
}
```

The general idea is that we don't have to know that `_getGCPRangeSize` is not supposed to be called only with bigger ranges, as the logic to decide what is a big range is itself in the function. Now I may miss some logic which makes this not practical, or maybe it does not matter so much, so feel free to ignore!

Reply:
Okay, that makes sense. We can view the `_getRangeSize` as the "generic" one and `_getGCPRangeSize` as the special case. I prefer less boolean params being passed around so this works for me. 

Updated.